evm/src/main/java/org/hyperledger/besu/evm/operation/SelfDestructOperation.java

```diff
@@ -60,19 +75,22 @@ public OperationResult execute(final MessageFrame frame, final EVM evm) {
     final Address address = frame.getRecipientAddress();
     final MutableAccount account = frame.getWorldUpdater().getAccount(address).getMutable();
 
-    frame.addSelfDestruct(address);
+    if (!eip6780Semantics || frame.wasCreatedInTransaction(address)) {
+      frame.addSelfDestruct(address);
+    }
 
     final MutableAccount recipient =
```

Comment:
Think it would help the understanding of this PR to rename this to `beneficiary` (or even`selfDestructBeneficiary`) to match the unit test and [the spec](https://ethereum.github.io/execution-specs/autoapi/ethereum/shanghai/vm/instructions/system/index.html?highlight=selfdestruct#ethereum.shanghai.vm.instructions.system.selfdestruct).

Reply:
Settled on `beneficiary` and `originator`